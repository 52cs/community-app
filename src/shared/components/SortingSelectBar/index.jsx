import _ from 'lodash';
import PT from 'prop-types';
import React from 'react';
import Select from 'components/Select';
import ArrowIcon from 'assets/images/ico-arrow-down.svg';
import './style.scss';

export default function SortingSelectBar({
  onSelect, options, title, value,
}) {
  return (
    <div styleName="sortingBar">
      <h2 styleName="title">
        {title}
      </h2>
      {
        options ? (
          <div styleName="view-options-toggle-container">
            <Select
              searchable={false}
              clearable={false}
              options={options}
              onChange={item => onSelect(item.value)}
              value={value}
              placeholder="Select an option"
              arrowRenderer={ArrowIcon}
            />
          </div>
        ) : null
      }
    </div>
  );
}

SortingSelectBar.defaultProps = {
  onSelect: _.noop,
  options: null,
  title: '',
  value: null,
};

SortingSelectBar.propTypes = {
  onSelect: PT.func,
  options: PT.arrayOf(PT.shape()),
  title: PT.string,
  value: PT.shape(),
};
