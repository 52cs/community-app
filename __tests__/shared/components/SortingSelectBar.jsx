import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import SortingSelectBar from 'components/SortingSelectBar';

const mockData = {
  title: 'title',
  options: [
    {
      label: 'label',
      value: 'value',
    },
  ],
};

const mockData2 = {
  title: 'title',
};

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <SortingSelectBar {...mockData} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <SortingSelectBar {...mockData2} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
