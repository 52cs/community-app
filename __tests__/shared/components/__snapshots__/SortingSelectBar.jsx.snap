// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-SortingSelectBar-___style__sortingBar___3esqf"
>
  <h2
    className="src-shared-components-SortingSelectBar-___style__title___36eG5"
  >
    title
  </h2>
  <div
    className="src-shared-components-SortingSelectBar-___style__view-options-toggle-container___1N4rY"
  >
    <Select
      arrowRenderer={[Function]}
      clearable={false}
      onChange={[Function]}
      options={
        Array [
          Object {
            "label": "label",
            "value": "value",
          },
        ]
      }
      placeholder="Select an option"
      searchable={false}
      selectRef={[Function]}
      value={null}
    />
  </div>
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="src-shared-components-SortingSelectBar-___style__sortingBar___3esqf"
>
  <h2
    className="src-shared-components-SortingSelectBar-___style__title___36eG5"
  >
    title
  </h2>
</div>
`;
